sway-core/src/declaration_engine/declaration_engine.rs

```diff
@@ -158,26 +211,69 @@ impl DeclarationEngine {
         self.slab.get(*index).expect_struct(span)
     }
 
-    fn de_add_monomorphized_struct_copy(
+    pub(crate) fn de_get_monomorphized_struct(
```

Comment:
I recommend making the `de_get_monomorphized_struct` and `de_get_monomorphized_function`, etc, functions all one function `de_get_monomorphized_version` that takes `original_id: DeclarationId,` and returns `Option<T> where T: MonomorphizeHelper`